lib/components/progress_hud/progress_hud.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_platform_widgets/flutter_platform_widgets.dart';
+import 'package:modal_progress_hud/modal_progress_hud.dart';
+
+class ProgressHUD extends StatelessWidget {
+  final Widget child;
+  final bool isShow;
+  const ProgressHUD({
+    required this.child,
+    this.isShow = false,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ModalProgressHUD(
+      inAsyncCall: isShow,
+      progressIndicator: Container(
+        width: 100,
+        height: 100,
+        decoration: const BoxDecoration(
+            color: Colors.white,
+            borderRadius: BorderRadius.all(Radius.circular(12))),
+        child: PlatformCircularProgressIndicator(
+          cupertino: (_, __) => CupertinoProgressIndicatorData(radius: 20),
```

Comment:
I have a question here: if from the parent widget, we are trying to differentiate the native behavior (which is not required from the requirement now) like [this](https://github.com/markgravity/flutter-survey/pull/34/files#diff-b356bb265b5af78c993fb53883c63112f03955871aa3719585bb45a21bf8fac0R20) then should you consider to continue on bringing the native feel to this Loading HUD widget too? This is a very iOS native loading HUD and when bringing it to material, it doesn't make sense imo. Unless you just use Scaffold from the beginning no?

Reply:
I'm not sure get your point here. But I'm using `PlatformCircularProgressIndicator` so it will switch to native UI base on the current platform.

Hope it can answer correct your concern 😁

Reply:
oh I see what you meant there, could you please update a quick glance to android too? As we are working on cross-platform, we need to recheck the behavior on both sides on some specific cases like the UI native feels imo.

Reply:
oh I see what you meant there, could you please update a quick glance to android too? As we are working on cross-platform, we need to recheck the behavior on both sides on some specific cases like the UI native feels imo.